### PR TITLE
EFI Prekernel (1/N): Kernel: Move boot info variables into a shared struct

### PR DESCRIPTION
**Pull request by spholz**
Instead of having multiple boot info variables, move them all into a common global BootInfo struct 'g_boot_info'.
Also make BootInfo more boot method-agnostic by moving multiboot-specific data into Multiboot1BootInfo, which is a member of the BootInfo::boot_method_specific union.

~~This depends on #24920, therefore draft.~~

### CODE REVIEW

File: Kernel/Arch/aarch64/MMU.cpp [hunk 1]
```diff
@@ -279,37 +279,42 @@ static u64* get_page_directory_table(u64* root_table, VirtualAddress virtual_add
...
     for (size_t o = 0; o < fdt_header->totalsize; o += 1) {
         // FIXME: Maybe increase the IO size here
         adjust_by_mapping_base(DeviceTree::s_fdt_storage)[o] = fdt_storage[o];
     }
 
-    *adjust_by_mapping_base(&physical_to_virtual_offset) = calculate_physical_to_link_time_address_offset();
-    *adjust_by_mapping_base(&kernel_mapping_base) = KERNEL_MAPPING_BASE;
-    *adjust_by_mapping_base(&kernel_load_base) = KERNEL_MAPPING_BASE;
+    *adjust_by_mapping_base(&g_boot_info.boot_method) = BootMethod::Multiboot1;
+    adjust_by_mapping_base(&g_boot_info.boot_method_specific)->pre_init.~PreInitBootInfo();
```
[the remark this answers is not in the document]
Comment (nico) [reply Kernel/Arch/aarch64/MMU.cpp:311]: Should we static_assert that g_boot_info.boot_method is_trivial? Then this dtor call does nothing and isn't needed, and the placement new isn't needed either.

[the remark this answers is not in the document]
Comment (spholz) [reply Kernel/Arch/aarch64/MMU.cpp:311]: `static_assert<IsTrivial<BootMethodSpecificBootInfo>>` fails if that is what you meant. I think it's not trivial due to the initializers in `Multiboot1BootInfo`?

[the remark this answers is not in the document]
Comment (spholz) [reply Kernel/Arch/aarch64/MMU.cpp:311]: But `static_assert<IsTrivial<PreInitBootInfo>>` succeeds. Does that mean the `~PreInitBootInfo` isn't needed?

Why do you think the placement new is unnecessary?

[the remark this answers is not in the document]
Comment (nico) [reply Kernel/Arch/aarch64/MMU.cpp:311]: It basically just memcpy()s, right? Not super important I suppose.

**Pull request merged**